_tests_/ship.test.js

```diff
@@ -0,0 +1,62 @@
+const Ship = require('../src/ship.js');
+
+describe("Ship", () => {
+  describe("with ports and itinerary", () => {
+    let london;
+    let liverpool;
+    let itinerary;
+    let ship;
+
+    beforeEach(() => {
+      london = { name: "London", ships: [], addShip: jest.fn(), removeShip: jest.fn() };
+      liverpool = { name: "Liverpool", ships: [], addShip: jest.fn(), removeShip: jest.fn() };
+      itinerary = { ports: [london, liverpool] };
+      ship = new Ship(itinerary);
+    });
+
+    it("can be instantiated", () => {
+      expect(ship).toBeInstanceOf(Object);
+    });
+
+    it("sets previous port as null", () => {
+      expect(ship.previousPort).toEqual(null);
+    });
+
+    it("has a starting port", () => {
+      expect(ship.currentPort).toBe(london);
+    });
+
+    it("gets added to port on instantiation", () => {
+      expect(london.addShip).toHaveBeenCalledWith(ship);
+    }); 
+
+    it("can set sail", () => {
+      ship.setSail();
+      expect(ship.currentPort).toBeFalsy();
+      expect(london.removeShip).toHaveBeenCalledWith(ship);
+    });
+
+    it("can dock at different port", () => {
+      ship.setSail();
+      ship.dock();
+      expect(ship.currentPort).toBe(liverpool);
+      expect(liverpool.addShip).toHaveBeenCalledWith(ship);
+    });
+  });
+});
+
+describe("setSail", () => {
```

Comment:
Here you took setSail out from the describe block, which is okay but you did not separate dock method.
I would suggest to separate both of them or leave both in the same Ship block, just for better consistency.